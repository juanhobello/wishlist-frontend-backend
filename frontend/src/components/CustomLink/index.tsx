import { ReactNode } from 'react';
import style from './CustomLink.module.css'
import { Link } from "react-router-dom";

interface CustomLinkProps {
  path: string;
  children: ReactNode;
}

export default function CustomLink({ path, children }: CustomLinkProps) {
  return (
    <Link className={style["link"]} to={path}>
      {children}
    </Link>
  )
}
