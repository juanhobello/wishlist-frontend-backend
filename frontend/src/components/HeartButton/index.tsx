import { FiHeart } from 'react-icons/fi'
import type { ButtonHTMLAttributes } from 'react'
import style from './HeartButton.module.css'

interface HeartButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  isActive: boolean
}

export default function HeartButton({ isActive, ...rest }: HeartButtonProps) {
  return (
    <button
      {...rest}
      className={`${style["heart-button"]} ${isActive ? style[".selected"] : ''}`}
      style={{ backgroundColor: isActive ? 'var(--heart-color)' : 'var(--gray-500)' }}
    >
      <FiHeart fontSize={20} />
    </button>
  )
}
