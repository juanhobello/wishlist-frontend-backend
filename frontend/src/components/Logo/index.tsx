import { Link } from "react-router-dom";
import styles from './Logo.module.css'

interface LogoProps {
  path: string;
}
export default function Logo({ path }: LogoProps) {
  return (
    <Link to={path} className={styles["logo-container"]} aria-label="Logo">
      <div className={styles["logo"]} />
    </Link>
  )
}
