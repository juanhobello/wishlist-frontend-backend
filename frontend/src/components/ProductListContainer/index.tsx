import type { ReactNode } from "react"
import style from './ProductListContainer.module.css'

interface ProductListContainerProps {
  children: ReactNode
}
export default function ProductListContainer({ children }: ProductListContainerProps) {
  return (
    <div className={style["products-list"]}>{children}</div>
  )
}
