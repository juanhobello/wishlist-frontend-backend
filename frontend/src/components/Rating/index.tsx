import styles from './Rating.module.css';
import Star from './Star';

interface RatingProps {
  rating: number
}

export default function Rating({ rating }: RatingProps) {

  const stars = Array.from({ length: 5 }, (_, i) => {
    const percentage = Math.min(Math.max(rating - i, 0), 1) * 100;
    return <Star key={i} percentage={percentage} />;
  });

  return (
    <div className={styles.rating}>
      {stars}
      <p>{rating.toFixed(1)}</p>
    </div>
  )
}

