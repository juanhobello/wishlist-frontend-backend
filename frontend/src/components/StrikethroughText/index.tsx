import type{ ReactNode } from "react"
import style from './StrikethroughText.module.css'

interface StrikethroughTextProps {
  children: string | ReactNode
}

export default function StrikethroughText({ children }: StrikethroughTextProps) {
  return (
    <s className={style["text"]}>{children}</s>
  )
}
