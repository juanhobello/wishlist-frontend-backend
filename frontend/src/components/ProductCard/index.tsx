import { Rating, StrikethroughText } from "@/components"
import type { ReactNode } from "react"
import styles from "./ProductCard.module.css"
import { Product } from "@/redux/types"
import { formatPrice } from "@/utils"

interface ProductCardProps {
  product: Product
  floatButton?: ReactNode
}
export default function ProductCard({ product, floatButton }: ProductCardProps) {
  return (
    <div className={styles["product-card"]}>
      {!!floatButton && <span className={styles["float-button"]}>{floatButton}</span>}

      <img src={product.image} className={styles["img"]} alt={product.name} />
     
      <div className={styles["title-product"]}>
        <p>{product.name}</p>
      </div>
     
      <Rating rating={product.rating} />

      <StrikethroughText>{formatPrice(Number(product.fullPriceInCents))}</StrikethroughText>

      <span className={styles["price"]}>
        {formatPrice(Number(product.salePriceInCents))}
      </span>
    </div>
  )
}
