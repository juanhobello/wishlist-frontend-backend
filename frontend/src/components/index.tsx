import Logo from "./Logo";
import CustomLink from './CustomLink'
import { Menu, MenuItem } from "./Menu";
import { Breadcrumbs, BreadcrumbItem } from "./Breadcrumbs";
import Divider from "./Divider";
import Rating from "./Rating";
import StrikethroughText from './StrikethroughText'
import ProductCard from "./ProductCard";
import ProductListContainer from "./ProductListContainer";
import HeartButton from './HeartButton';

export {
  Logo,
  CustomLink,
  Menu,
  MenuItem,
  Breadcrumbs,
  BreadcrumbItem,
  Divider,
  Rating,
  StrikethroughText,
  ProductCard,
  ProductListContainer,
  HeartButton
}