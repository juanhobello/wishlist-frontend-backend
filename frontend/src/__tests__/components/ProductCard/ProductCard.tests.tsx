import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import {ProductCard} from "@/components";

const productMock = {
  code: "123",
  name: "Product Name",
  image: "image-url.jpg",
  rating: 4.5,
  available: true,
  visible: true,
  fullPriceInCents: '5000',
  salePriceInCents: '4000',
  details: {
    name: "Product Name",
    description: "Description"
  },
  stockAvailable: true
};

describe("ProductCard component", () => {
  test("matches snapshot", () => {
    const { container } = render(<ProductCard product={productMock} />);
    expect(container).toMatchSnapshot();
  });

  test("renders product details correctly", () => {
    render(<ProductCard product={productMock} />);

    expect(screen.getByText(productMock.name)).toBeInTheDocument();

    const img = screen.getByAltText(productMock.name);
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", productMock.image);

    expect(screen.getByText("R$ 50,00")).toBeInTheDocument();
    expect(screen.getByText("R$ 40,00")).toBeInTheDocument();

    expect(screen.getByText("4.5")).toBeInTheDocument();
  });

  test("renders floatButton when provided", () => {
    render(
      <ProductCard
        product={productMock}
        floatButton={<button>Buy</button>}
      />
    );
    expect(screen.getByText("Buy")).toBeInTheDocument();
  });

  test("does not render floatButton when not provided", () => {
    render(<ProductCard product={productMock} />);
    expect(screen.queryByRole("button")).not.toBeInTheDocument();
  });
})
