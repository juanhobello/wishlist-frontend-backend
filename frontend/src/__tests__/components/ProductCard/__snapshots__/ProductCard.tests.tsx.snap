// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component matches snapshot 1`] = `
<div>
  <div
    class="product-card"
  >
    <img
      alt="Product Name"
      class="img"
      src="image-url.jpg"
    />
    <div
      class="title-product"
    >
      <p>
        Product Name
      </p>
    </div>
    <div
      class="rating"
    >
      <svg
        aria-label="Star"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            id="grad-100"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              offset="100%"
              style="stop-opacity: 1;"
            />
            <stop
              offset="100%"
              style="stop-opacity: 1;"
            />
          </lineargradient>
        </defs>
        <path
          d="M8 .192l1.889 5.812h6.111l-4.944 3.594 1.889 5.813-4.944-3.594-4.944 3.594 1.889-5.813-4.944-3.594h6.111z"
          fill="url(#grad-100)"
        />
      </svg>
      <svg
        aria-label="Star"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            id="grad-100"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              offset="100%"
              style="stop-opacity: 1;"
            />
            <stop
              offset="100%"
              style="stop-opacity: 1;"
            />
          </lineargradient>
        </defs>
        <path
          d="M8 .192l1.889 5.812h6.111l-4.944 3.594 1.889 5.813-4.944-3.594-4.944 3.594 1.889-5.813-4.944-3.594h6.111z"
          fill="url(#grad-100)"
        />
      </svg>
      <svg
        aria-label="Star"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            id="grad-100"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              offset="100%"
              style="stop-opacity: 1;"
            />
            <stop
              offset="100%"
              style="stop-opacity: 1;"
            />
          </lineargradient>
        </defs>
        <path
          d="M8 .192l1.889 5.812h6.111l-4.944 3.594 1.889 5.813-4.944-3.594-4.944 3.594 1.889-5.813-4.944-3.594h6.111z"
          fill="url(#grad-100)"
        />
      </svg>
      <svg
        aria-label="Star"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            id="grad-100"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              offset="100%"
              style="stop-opacity: 1;"
            />
            <stop
              offset="100%"
              style="stop-opacity: 1;"
            />
          </lineargradient>
        </defs>
        <path
          d="M8 .192l1.889 5.812h6.111l-4.944 3.594 1.889 5.813-4.944-3.594-4.944 3.594 1.889-5.813-4.944-3.594h6.111z"
          fill="url(#grad-100)"
        />
      </svg>
      <svg
        aria-label="Star"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <lineargradient
            id="grad-50"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              offset="50%"
              style="stop-opacity: 1;"
            />
            <stop
              offset="50%"
              style="stop-opacity: 1;"
            />
          </lineargradient>
        </defs>
        <path
          d="M8 .192l1.889 5.812h6.111l-4.944 3.594 1.889 5.813-4.944-3.594-4.944 3.594 1.889-5.813-4.944-3.594h6.111z"
          fill="url(#grad-50)"
        />
      </svg>
      <p>
        4.5
      </p>
    </div>
    <s
      class="text"
    >
      R$ 50,00
    </s>
    <span
      class="price"
    >
      R$ 40,00
    </span>
  </div>
</div>
`;
